### Non-coding RNA annotation

install infernal and easel from bioconda

In [9]:
%%bash
cd ../../Data/Rfam
wget ftp://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/Rfam.cm.gz
gunzip Rfam.cm.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/Rfam.clanin

--2019-05-06 10:21:06--  ftp://ftp.ebi.ac.uk/pub/databases/Rfam/CURRENT/Rfam.cm.gz
           => ‘Rfam.cm.gz’
Resolving ftp.ebi.ac.uk (ftp.ebi.ac.uk)... 193.62.192.4
Connecting to ftp.ebi.ac.uk (ftp.ebi.ac.uk)|193.62.192.4|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/databases/Rfam/CURRENT ... done.
==> SIZE Rfam.cm.gz ... 35670981
==> PASV ... done.    ==> RETR Rfam.cm.gz ... done.
Length: 35670981 (34M) (unauthoritative)

     0K .......... .......... .......... .......... ..........  0%  168K 3m28s
    50K .......... .......... .......... .......... ..........  0%  358K 2m32s
   100K .......... .......... .......... .......... ..........  0%  134M 1m41s
   150K .......... .......... .......... .......... ..........  0%  358K 1m40s
   200K .......... .......... .......... .......... ..........  0%  192M 80s
   250K .......... .......... .......... .......... ..........  0%  360K 83s
   300K ...

In [11]:
%%bash
cd ../../Data/Rfam
cmpress Rfam.cm

Working...    done.
Pressed and indexed 3016 CMs and p7 HMM filters (3016 names and 3016 accessions).
Covariance models and p7 filters pressed into binary file:  Rfam.cm.i1m
SSI index for binary covariance model file:                 Rfam.cm.i1i
Optimized p7 filter profiles (MSV part)  pressed into:      Rfam.cm.i1f
Optimized p7 filter profiles (remainder) pressed into:      Rfam.cm.i1p


In [19]:
%%bash
cd ../../Data/Rfam
cat ../../Data/Rhoto_IFO0880_4/Rhoto_IFO0880_4_AssemblyScaffolds.fasta \
    ../../Data/Rhoto_IFO0880_4/Rhoto_IFO0880_4_MitoAssemblyScaffolds.fasta \
    > Rhoto_IFO0880_4_AssemblyScaffolds_and_Mito.fasta

In [20]:
%%bash
cd ../../Data/Rfam
esl-seqstat Rhoto_IFO0880_4_AssemblyScaffolds_and_Mito.fasta

Format:              FASTA
Alphabet type:       DNA
Number of sequences: 31
Total # residues:    20794214
Smallest:            13957
Largest:             2216085
Average length:      670781.1


In [21]:
20794214*2/1000000

41.588428

In [22]:
%%bash
cd ../../Data/Rfam
cmscan --cpu 24 -Z 41.588428 --cut_ga --rfam --nohmmonly --tblout IFO0880_4.tblout --fmt 2 --clanin Rfam.clanin Rfam.cm \
    Rhoto_IFO0880_4_AssemblyScaffolds_and_Mito.fasta > IFO0880_4.cmscan

In [23]:
%%bash
cd ../../Data/Rfam
head IFO0880_4.tblout

#idx target name          accession query name           accession clan name mdl mdl from   mdl to seq from   seq to strand trunc pass   gc  bias  score   E-value inc olp anyidx afrct1 afrct2 winidx wfrct1 wfrct2 description of target
#--- -------------------- --------- -------------------- --------- --------- --- -------- -------- -------- -------- ------ ----- ---- ---- ----- ------ --------- --- --- ------ ------ ------ ------ ------ ------ ---------------------
1    tRNA                 RF00005   scaffold_1           -         CL00001    cm        1       71   553894   553822      -    no    1 0.60   0.0   69.2   2.2e-14  !   *       -      -      -      -      -      - tRNA
2    tRNA                 RF00005   scaffold_1           -         CL00001    cm        1       71   566953   567025      +    no    1 0.60   0.0   69.2   2.2e-14  !   *       -      -      -      -      -      - tRNA
3    tRNA                 RF00005   scaffold_1           -         CL00001    cm        1     

In [24]:
%%bash
cd ../../Data/Rfam
grep -v " = " IFO0880_4.tblout > IFO0880_4.deoverlapped.tblout

In [25]:
%%bash
cd ../../Data/Rfam
head IFO0880_4.deoverlapped.tblout

#idx target name          accession query name           accession clan name mdl mdl from   mdl to seq from   seq to strand trunc pass   gc  bias  score   E-value inc olp anyidx afrct1 afrct2 winidx wfrct1 wfrct2 description of target
#--- -------------------- --------- -------------------- --------- --------- --- -------- -------- -------- -------- ------ ----- ---- ---- ----- ------ --------- --- --- ------ ------ ------ ------ ------ ------ ---------------------
1    tRNA                 RF00005   scaffold_1           -         CL00001    cm        1       71   553894   553822      -    no    1 0.60   0.0   69.2   2.2e-14  !   *       -      -      -      -      -      - tRNA
2    tRNA                 RF00005   scaffold_1           -         CL00001    cm        1       71   566953   567025      +    no    1 0.60   0.0   69.2   2.2e-14  !   *       -      -      -      -      -      - tRNA
3    tRNA                 RF00005   scaffold_1           -         CL00001    cm        1     

In [26]:
%%bash
cd ../../Data/Rfam
wc -l *.tblout

   201 IFO0880_4.deoverlapped.tblout
   330 IFO0880_4.tblout
   531 total


In [112]:
# Edit it in excel to make a gff
# Only keep tRNA and rRNA for BOFdat and use 'GeneID=tRNA_x' format

In [27]:
%%bash
cd ../../Data/Rfam/
head IFO0880_4_Rfam.gff3

scaffold_1	Infernal	tRNA	553822	553894	.	-	.	GeneID=tRNA_1
scaffold_1	Infernal	tRNA	566953	567025	.	+	.	GeneID=tRNA_2
scaffold_1	Infernal	tRNA	601576	601648	.	+	.	GeneID=tRNA_3
scaffold_1	Infernal	tRNA	606803	606875	.	+	.	GeneID=tRNA_4
scaffold_1	Infernal	tRNA	716396	716467	.	-	.	GeneID=tRNA_5
scaffold_1	Infernal	tRNA	736681	736752	.	-	.	GeneID=tRNA_6
scaffold_1	Infernal	tRNA	1149316	1149413	.	+	.	GeneID=tRNA_7
scaffold_1	Infernal	tRNA	1412446	1412545	.	-	.	GeneID=tRNA_8
scaffold_1	Infernal	tRNA	1513931	1514030	.	+	.	GeneID=tRNA_9
scaffold_1	Infernal	tRNA	1928091	1928162	.	+	.	GeneID=tRNA_10


In [28]:
%%bash
cd ../../Data/Rhoto_IFO0880_4/
head Rhoto_IFO0880_4_GeneCatalog_20170509.gff3

##gff-version 3
##sequence-region scaffold_1 1 2216085
scaffold_1	prediction	gene	1571	2453	0	-	.	ID=gene_1;Name=jgi.p|Rhoto_IFO0880_4|8369;portal_id=Rhoto_IFO0880_4;proteinId=8369;transcriptId=8497
scaffold_1	prediction	mRNA	1571	2453	.	-	.	ID=mRNA_1;Name=jgi.p|Rhoto_IFO0880_4|8369;Parent=gene_1;proteinId=8369;track=FrozenGeneCatalog;transcriptId=8497
scaffold_1	prediction	exon	2283	2453	.	-	.	ID=exon_1_1;Parent=mRNA_1
scaffold_1	prediction	CDS	2283	2453	.	-	0	ID=CDS_1;Parent=mRNA_1
scaffold_1	prediction	exon	1953	2165	.	-	.	ID=exon_1_2;Parent=mRNA_1
scaffold_1	prediction	CDS	1953	2165	.	-	0	ID=CDS_1;Parent=mRNA_1
scaffold_1	prediction	exon	1770	1895	.	-	.	ID=exon_1_3;Parent=mRNA_1
scaffold_1	prediction	CDS	1770	1895	.	-	0	ID=CDS_1;Parent=mRNA_1


In [130]:
%%bash
cd ../../Data/Rhoto_IFO0880_4/
grep $'\tmRNA' Rhoto_IFO0880_4_GeneCatalog_20170509.gff3 | wc -l
grep $'\tmRNA' Rhoto_IFO0880_4_GeneCatalog_20170509.gff3 | head

8490
scaffold_1	prediction	mRNA	1571	2453	.	-	.	ID=mRNA_1;Name=jgi.p|Rhoto_IFO0880_4|8369;Parent=gene_1;proteinId=8369;track=FrozenGeneCatalog;transcriptId=8497
scaffold_1	prediction	mRNA	3233	7809	.	-	.	ID=mRNA_2;Name=jgi.p|Rhoto_IFO0880_4|8370;Parent=gene_2;proteinId=8370;track=FrozenGeneCatalog;transcriptId=8498
scaffold_1	prediction	mRNA	8155	10023	.	+	.	ID=mRNA_3;Name=jgi.p|Rhoto_IFO0880_4|8371;Parent=gene_3;proteinId=8371;track=FrozenGeneCatalog;transcriptId=8499
scaffold_1	prediction	mRNA	10143	11954	.	+	.	ID=mRNA_4;Name=jgi.p|Rhoto_IFO0880_4|8372;Parent=gene_4;proteinId=8372;track=FrozenGeneCatalog;transcriptId=8500
scaffold_1	prediction	mRNA	12020	13736	.	+	.	ID=mRNA_5;Name=jgi.p|Rhoto_IFO0880_4|8373;Parent=gene_5;proteinId=8373;track=FrozenGeneCatalog;transcriptId=8501
scaffold_1	prediction	mRNA	13861	19221	.	-	.	ID=mRNA_6;Name=jgi.p|Rhoto_IFO0880_4|8374;Parent=gene_6;proteinId=8374;track=FrozenGeneCatalog;transcriptId=8502
scaffold_1	prediction	mRNA	20070	23052	.	+	.	ID=mRNA

In [134]:
%%bash
cd ../../Data/Rhoto_IFO0880_4/
grep $'\tmRNA' Rhoto_IFO0880_4_GeneCatalog_20170509.gff3 | sed 's/proteinId/GeneID/' | head
grep $'\tmRNA' Rhoto_IFO0880_4_GeneCatalog_20170509.gff3 | sed 's/proteinId/GeneID/' > IFO0880_4_JGI.gff3
mv IFO0880_4_JGI.gff3 ../Rfam/

scaffold_1	prediction	mRNA	1571	2453	.	-	.	ID=mRNA_1;Name=jgi.p|Rhoto_IFO0880_4|8369;Parent=gene_1;GeneID=8369;track=FrozenGeneCatalog;transcriptId=8497
scaffold_1	prediction	mRNA	3233	7809	.	-	.	ID=mRNA_2;Name=jgi.p|Rhoto_IFO0880_4|8370;Parent=gene_2;GeneID=8370;track=FrozenGeneCatalog;transcriptId=8498
scaffold_1	prediction	mRNA	8155	10023	.	+	.	ID=mRNA_3;Name=jgi.p|Rhoto_IFO0880_4|8371;Parent=gene_3;GeneID=8371;track=FrozenGeneCatalog;transcriptId=8499
scaffold_1	prediction	mRNA	10143	11954	.	+	.	ID=mRNA_4;Name=jgi.p|Rhoto_IFO0880_4|8372;Parent=gene_4;GeneID=8372;track=FrozenGeneCatalog;transcriptId=8500
scaffold_1	prediction	mRNA	12020	13736	.	+	.	ID=mRNA_5;Name=jgi.p|Rhoto_IFO0880_4|8373;Parent=gene_5;GeneID=8373;track=FrozenGeneCatalog;transcriptId=8501
scaffold_1	prediction	mRNA	13861	19221	.	-	.	ID=mRNA_6;Name=jgi.p|Rhoto_IFO0880_4|8374;Parent=gene_6;GeneID=8374;track=FrozenGeneCatalog;transcriptId=8502
scaffold_1	prediction	mRNA	20070	23052	.	+	.	ID=mRNA_7;Name=jgi.p|Rhoto_IFO

In [31]:
%%bash
cd ../../Data/Rfam/
cat IFO0880_4_JGI.gff3 IFO0880_4_Rfam.gff3 > IFO0880_4_JGI_Rfam.gff3

In [32]:
from Bio import SeqIO
from Bio.Alphabet import generic_dna
from BCBio import GFF

gff_file = '../../Data/Rfam/IFO0880_4_JGI_Rfam.gff3'
fasta_file = '../../Data/Rfam/Rhoto_IFO0880_4_AssemblyScaffolds_and_Mito.fasta'
out_file = '../../Data/Rfam/IFO0880_4_JGI_Rfam.gb'
fasta_input = SeqIO.to_dict(SeqIO.parse(fasta_file, "fasta", generic_dna))
gff_iter = GFF.parse(gff_file, fasta_input)
SeqIO.write(gff_iter, out_file, "genbank")

31

In [33]:
%%bash
cd ../../Data/Rfam/
head IFO0880_4_JGI_Rfam.gb

LOCUS       scaffold_1           2216085 bp    DNA              UNK 01-JAN-1980
DEFINITION  scaffold_1.
ACCESSION   scaffold_1
VERSION     scaffold_1
KEYWORDS    .
SOURCE      .
  ORGANISM  .
            .
FEATURES             Location/Qualifiers
     tRNA            complement(553822..553894)


IFO0880_4_JGI_Rfam.gb is used in BOFdat to estimate RNA composition

### Mitochondrial genome annotation

download mitochondria protein sequences from http://megasun.bch.umontreal.ca/People/lang/FMGP/proteins.html  
Rhoto_IFO0880_3 has mitochondrial genes and annotation included in the genome assembly  

In [66]:
%%bash
cd ../../Data/Mito/
head Mito_Proteins.fa

>cob Malawimonas jakobiformis
MVGMYPVPTTISYYWSFGSLAGLCLVIQIVTGILLAMHYTPNINLAFISVEHIMRDVNYG
WLLRYIHANGASMFFIVLYLHMFRGLYYGSYLYPRQMVWSIGVIIFLATMATAFMGYVLP
WGQMSFWGATVITNFFSAFPIVGESIVHWLWGGFSVDNATLNRFFSLHYFLPFIIAALSG
LHLAILHIPKANNPMGIRTLNDYLQFHPYFTVKDIFSLVIFIVIFAIFVFFYPNILGHAD
NYIQANPLVTPAHIVPEWYFLPFYAILRAIPDKLGGVLVMFAAIVILLFIPLLNISNVRS
SWFRPLFRPFFWLLVVDSLILAWIGGNTAEAPYVQIGQIATFFYFFYFLFLIPILAKIES
LINKY*

>cob Reclinomonas americana 


In [69]:
%%bash
cd ../../Data/Mito/
reformat.sh in=Mito_Proteins.fa out=Mito_Proteins_Fixed.fa overwrite=true addunderscore ignorejunk

java -ea -Xmx200m -cp /users/joonhoonkim/anaconda3/opt/bbmap-38.22-0/current/ jgi.ReformatReads in=Mito_Proteins.fa out=Mito_Proteins_Fixed.fa overwrite=true addunderscore ignorejunk
Executing jgi.ReformatReads [in=Mito_Proteins.fa, out=Mito_Proteins_Fixed.fa, overwrite=true, addunderscore, ignorejunk]

Input is being processed as unpaired
Input:                  	227 reads          	73320 bases
Output:                 	227 reads (100.00%) 	73320 bases (100.00%)

Time:                         	0.233 seconds.
Reads Processed:         227 	0.97k reads/sec
Bases Processed:       73320 	0.31m bases/sec


In [71]:
%%bash
cd ../../Data/Mito/
cat Mito_Proteins_Fixed.fa

>cob_Malawimonas_jakobiformis
MVGMYPVPTTISYYWSFGSLAGLCLVIQIVTGILLAMHYTPNINLAFISVEHIMRDVNYGWLLRYIHANG
ASMFFIVLYLHMFRGLYYGSYLYPRQMVWSIGVIIFLATMATAFMGYVLPWGQMSFWGATVITNFFSAFP
IVGESIVHWLWGGFSVDNATLNRFFSLHYFLPFIIAALSGLHLAILHIPKANNPMGIRTLNDYLQFHPYF
TVKDIFSLVIFIVIFAIFVFFYPNILGHADNYIQANPLVTPAHIVPEWYFLPFYAILRAIPDKLGGVLVM
FAAIVILLFIPLLNISNVRSSWFRPLFRPFFWLLVVDSLILAWIGGNTAEAPYVQIGQIATFFYFFYFLF
LIPILAKIESLINKY*
>cob_Reclinomonas_americana_
MRLLKRPLIRELNSFIVDYPTPSNLSYWWNFGFIAAFCLVVQIATGIFLAMHYTPHVDLAFISVEHIMRD
VNYGWLMRYIHANGASMFFIAVYIHMFRGLYYGSYASPREFLWIIGVIILLIMILTAFMGYVLPWGQMSF
WGATVITNLASAVPYIGEHIVYWLWGGFSVDNATLNRFYSLHYLFPFIIAGLVGLHLIVLHEDGSNNPLG
IDSKVDKIPFYPYYYVKDLFGIVVFGLFFSVFVFFYPNLLGHPDNYIEANPLVTPPHIVPEWYFLPFYAI
LRSIPNKLLGVIAMLVSILILIALPFLNTSEIRSSQFRPIHRKLFWLFFVDCVILGWIGGNVPETPYLEI
GQIATVFYFAYFLVFIPLLGKLESYLINLDTNQYQKHLVK*
>cob_Rickettsia_prowazekii
MNKEIIHKKS NGIIEWIDYR LPIFSFLKHF SYYQTPKNLN YLWNLGSIAGIALVIQIITG VILAM
HYTPH VDHAFESVER IMRNVNYGWL LRYTHAVGASMFFAAIYLHI ARGLYYGSYK TPRELLWHIG
 II

In [94]:
%%bash
cd ../../Data/Rhoto_IFO0880_3/
head Rhoto_IFO0880_3_all_proteins_20160906.aa.fasta

>jgi|Rhoto_IFO0880_3|921252|estExt_Genewise1.C_1_t10001
MNRLFGSSSSKPKPSLADAIASTDLRVDSIEVKIRKLDAELTKYRDQLKKMRDGPGKNAVQQRALRVLKQ
KKLYESQIAQLQQQSFNMEQASMTTENLRNTMATVDAMKTANKEMKKQYGKIDIDEIEAMHDDMADLLDS
ANDVQEAMSRTYGVPEEVDEADLEAELEALGNEFEEEEGIPSYLQADATSELPDFVDEAPQKEEAPKERL
GEGVV*
>jgi|Rhoto_IFO0880_3|921253|estExt_Genewise1.C_1_t10002
MNRLFGSSSSKPKPSLADAIASTDLRVDSIEVKIRKLDAELTKYRDQLKKMRDGPGKNAVQQRALRVLKQ
KKLYESQIAQLQQQSFNMEQASMTTENLRNTMATVDAMKTANKEMKKQYGKIDIDEIEAMHDDMADLLDS
ANDVQEAMSRTYGVPEEVDEADLEAELEALGNEFEEEEGIPSYLQADATSELPDFVDEAPQKEEAPKERL
GEGVV*


In [95]:
%%bash
cd ../../Data/Rhoto_IFO0880_3/
cat Rhoto_IFO0880_3_all_proteins_20160906.aa.fasta | sed 's/jgi|Rhoto_IFO0880_3|//g' | sed 's/|/ /g' \
    > Rhoto_IFO0880_3_all_proteins_20160906.aa.modified.fasta
samtools faidx Rhoto_IFO0880_3_all_proteins_20160906.aa.modified.fasta
makeblastdb -in Rhoto_IFO0880_3_all_proteins_20160906.aa.modified.fasta -parse_seqids -dbtype prot -out Rhoto_IFO0880_3_All



Building a new DB, current time: 07/26/2019 14:19:34
New DB name:   /mnt/san-nfs/nfs/users/joonhoonkim/Data/R_toruloides/Rhoto_IFO0880_3/Rhoto_IFO0880_3_All
New DB title:  Rhoto_IFO0880_3_all_proteins_20160906.aa.modified.fasta
Sequence type: Protein
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 164153 sequences in 7.53851 seconds.


In [97]:
%%bash
cd ../../Data/Mito/
blastp -query Mito_Proteins_Fixed.fa -db ../../Data/Rhoto_IFO0880_3/Rhoto_IFO0880_3_All \
    -outfmt "6 std qcovs" -num_threads 24 > Mito_Proteins_to_IFO0880_3.txt

FASTA-Reader: Ignoring invalid residues at position(s): On line 1060: 17-19


In [98]:
%%bash
cd ../../Data/Mito/
head Mito_Proteins_to_IFO0880_3.txt

cob_Malawimonas_jakobiformis	979592	54.417	283	112	6	37	308	1	277	4.21e-89	271	74
cob_Malawimonas_jakobiformis	979594	53.333	285	116	6	37	308	1	281	3.98e-88	269	74
cob_Malawimonas_jakobiformis	950128	46.784	342	120	5	22	360	11	293	9.75e-88	268	93
cob_Malawimonas_jakobiformis	946001	46.784	342	120	5	22	360	11	293	9.75e-88	268	93
cob_Malawimonas_jakobiformis	900628	57.031	128	55	0	181	308	1	128	1.03e-45	154	35
cob_Malawimonas_jakobiformis	900627	70.000	80	24	0	37	116	1	80	1.73e-33	121	22
cob_Reclinomonas_americana_	979594	53.684	285	115	6	51	322	1	281	8.18e-91	276	70
cob_Reclinomonas_americana_	979592	54.064	283	113	6	51	322	1	277	1.23e-90	276	70
cob_Reclinomonas_americana_	950128	45.892	353	129	5	34	383	9	302	1.87e-90	276	90
cob_Reclinomonas_americana_	946001	45.892	353	129	5	34	383	9	302	1.87e-90	276	90


In [41]:
import csv
import numpy as np
import matplotlib.pyplot as plt

In [101]:
# Parse blastp results
blast = "../../Data/Mito/Mito_Proteins_to_IFO0880_3.txt"
gene_list = set()
match_list = {}

min_iden = 50
min_eval = 1e-10

with open(blast,'r') as f:
    for line in csv.reader(f,delimiter='\t'):
        query = line[0]
        subject = line[1:]
        gene_list.add(query)
        if (float(subject[1]) > min_iden) and (float(subject[9]) < min_eval):
            if query in match_list:
                match_list[query].append(subject)
            else:
                match_list[query] = [subject]

print(len(gene_list),len(match_list))

201 143


In [102]:
gene_list

{'Capsaspora_owczsarsaki_atp6_;_258_aa_on_',
 'Capsaspora_owczsarsaki_atp9_;_74_aa_on_',
 'Capsaspora_owczsarsaki_cob_;_381_aa_on_',
 'Capsaspora_owczsarsaki_cox1_;_565_aa_on_',
 'Capsaspora_owczsarsaki_cox2_;_277_aa_on_',
 'Capsaspora_owczsarsaki_cox3_;_264_aa_on_',
 'Capsaspora_owczsarsaki_nad1_;_331_aa_on_',
 'Capsaspora_owczsarsaki_nad2_;_543_aa_on_',
 'Capsaspora_owczsarsaki_nad3_;_117_aa_on_',
 'Capsaspora_owczsarsaki_nad4L_;_99_aa_on_',
 'Capsaspora_owczsarsaki_nad4_;_477_aa_on_',
 'Capsaspora_owczsarsaki_nad5_;_662_aa_on_',
 'Capsaspora_owczsarsaki_nad6_;_198_aa_on_',
 'Capsaspora_owczsarsaki_orf157_;_157_aa_on_',
 'Capsaspora_owczsarsaki_orf165_;_165_aa_on_',
 'Capsaspora_owczsarsaki_orf190_;_190_aa_on_',
 "Capsaspora_owczsarsaki_orf233_;_251_aa_on_'nu6687iac329_mt'",
 "Capsaspora_owczsarsaki_orf280_;_280_aa_on_'nu6687iac329_mt'",
 'Capsaspora_owczsarsaki_rpl14_;_127_aa_on_',
 'Capsaspora_owczsarsaki_rpl16_;_184_aa_on_',
 'Capsaspora_owczsarsaki_rpl2_;_483_aa_on_',
 "Capsaspor

In [121]:
Mito_proteinId = dict()
for k, v in match_list.items():
    #if not any(x in v[0][0] for x in Mito_proteinId.values()):
    temp = k.split('_')
    x = [y for y in temp if any(z in y for z in ['cob','cox','atp','nad'])]
    temp = Mito_proteinId.setdefault(x[0], dict())
    temp[v[0][0]] = temp.setdefault(v[0][0],float()) + float(v[0][1])
    Mito_proteinId[x[0]] = temp
    print(x[0], v[0][0:2])

cob ['979592', '54.417']
cob ['979594', '53.684']
cob ['979592', '52.297']
cob ['979592', '56.184']
cob ['979592', '51.246']
cob ['950128', '50.386']
cob ['979592', '52.448']
cob ['979592', '56.228']
cob ['979592', '54.064']
cob ['979592', '52.650']
cob ['979592', '53.710']
cob ['979592', '55.516']
cob ['900628', '54.762']
cob ['950128', '54.155']
cob ['950128', '57.990']
cob ['950128', '52.895']
cob ['979592', '59.364']
cob ['900628', '53.968']
cob ['900628', '50.781']
cob ['900627', '63.750']
cob ['900627', '57.500']
cox1 ['900659', '64.655']
cox1 ['950130', '52.727']
cox1 ['950130', '52.000']
cox1 ['950130', '56.998']
cox1 ['950130', '52.174']
cox1 ['950130', '55.624']
cox1 ['950130', '57.631']
cox1 ['950130', '55.985']
cox1 ['950130', '57.169']
cox1 ['900659', '66.667']
cox1 ['950130', '54.495']
cox1 ['950130', '53.775']
cox1 ['950130', '59.450']
cox1 ['950130', '60.886']
cox1 ['950130', '52.459']
cox1 ['950130', '59.817']
cox1 ['950130', '62.637']
cox1 ['950130', '61.792']
cox1 ['

In [126]:
Mito_proteinId

{'cob': {'979592': 655.721,
  '979594': 106.666,
  '950128': 436.87,
  '900628': 159.511,
  '900627': 121.25},
 'cox1': {'900659': 328.382, '950130': 1626.5469999999998},
 'cox2': {'950130': 1083.48, '518258': 339.301},
 'cox3': {'946002': 903.938, '889639': 269.195, '900631': 170.958},
 'atp9': {'900663': 432.82000000000005},
 'nad1': {'900641': 280.145, '950129': 58.861},
 'nad3': {'863320': 52.874, '879368': 56.989, '946000': 50.515},
 'nad4': {'900607': 279.053},
 'nad4L': {'518739': 55.422,
  '945993': 72.093,
  '861805': 63.953,
  '866713': 66.279},
 'nad5': {'900668': 247.29200000000003, '945994': 53.959},
 'atp6': {'900623': 51.984, '945998': 51.634},
 'atp8': {'900650': 110.42099999999999}}

In [179]:
targets = set(sum([list(x.keys()) for x in Mito_proteinId.values()],[]))
targets

{'518258',
 '518739',
 '861805',
 '863320',
 '866713',
 '879368',
 '889639',
 '900607',
 '900623',
 '900627',
 '900628',
 '900631',
 '900641',
 '900650',
 '900659',
 '900663',
 '900668',
 '945993',
 '945994',
 '945998',
 '946000',
 '946002',
 '950128',
 '950129',
 '950130',
 '979592',
 '979594'}

In [136]:
%%bash
cd ../../Data/Rhoto_IFO0880_3/
head Rhoto_IFO0880_3_all_proteinId_transcriptId_map_20160906.txt

proteinId	transcriptId
6	157
7	158
14	165
26	177
37	188
45	196
51	202
57	208
69	220


In [180]:
# Parse blastp results
temp = "../../Data/Rhoto_IFO0880_3/Rhoto_IFO0880_3_all_proteinId_transcriptId_map_20160906.txt"
mapping = {}

with open(temp,'r') as f:
    for line in csv.reader(f,delimiter='\t'):
        query = line[0]
        subject = line[1]
        mapping[query] = subject

In [152]:
','.join([mapping[x] for x in targets])

'900810,900782,900778,866864,950280,900774,900801,518890,950279,946153,979743,863471,900779,950281,946144,861956,518409,946151,879519,900758,946149,900814,900819,946145,889790,979745,900792'

In [145]:
%%bash
cd ../../Data/Rhoto_IFO0880_3/
head Rhoto_IFO0880_3_all_transcripts_20160906.nt.fasta

>jgi|Rhoto_IFO0880_3|921403|estExt_Genewise1.C_1_t10001
CACCTCTGACTGCGGTTCGAAGGTCTCTTACTAAGGTCTACAGCGTTCCTGCAGTACCAATTTCGTGCTG
TGCGGGCTCTGCGAAGCTGTAGATCGCGCCGACGTCGCCCTCGAGTTTCCACTCTCCCTTCTGCTTCTGC
CCATCTCGCTCACCTCGCTCTTCTTCGTCGCAACCGCTGCATATAGAGCTTGGACGACTCAGCGGCTGCC
TGCTACCCCTCGCACGCCAATCACTTACCTTCACGCCGGTCTCTTCCTGCCCCACCTCGACTCGGACTTC
TCGCACGCCTCGCAATCCGCTCGCACGCGTTCTCTGACAGACCAGAATCCGTCGGAGAGTATCTGAAGAT
CGACCGGCTTCCCAGTCTTACTGCCACGCACTCTCGCCACCCTCTCGCTTCGACCTTCGCTTGCCGGCCC
GCCATGAACCGCCTCTTCGGCTCGTCGTCCTCAAAACCGAAACCCTCTCTCGCCGACGCAATCGCTTCCA
CCGACCTCCGCGTCGACTCGATCGAGGTCAAGATCCGCAAGCTCGATGCAGAGCTCACCAAGTACCGCGA
CCAGCTGAAAAAGATGCGGGACGGGCCAGGGAAGAACGCCGTCCAGCAGCGCGCCCTCCGCGTCCTCAAG


In [146]:
%%bash
cd ../../Data/Rhoto_IFO0880_3/
cat Rhoto_IFO0880_3_all_transcripts_20160906.nt.fasta | sed 's/jgi|Rhoto_IFO0880_3|//g' | sed 's/|/ /g' \
    > Rhoto_IFO0880_3_all_transcripts_20160906.nt.modified.fasta
samtools faidx Rhoto_IFO0880_3_all_transcripts_20160906.nt.modified.fasta
makeblastdb -in Rhoto_IFO0880_3_all_transcripts_20160906.nt.modified.fasta -parse_seqids -dbtype nucl -out Rhoto_IFO0880_3_All_transcripts



Building a new DB, current time: 07/26/2019 14:54:59
New DB name:   /mnt/san-nfs/nfs/users/joonhoonkim/Data/R_toruloides/Rhoto_IFO0880_3/Rhoto_IFO0880_3_All_transcripts
New DB title:  Rhoto_IFO0880_3_all_transcripts_20160906.nt.modified.fasta
Sequence type: Nucleotide
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 164153 sequences in 11.0689 seconds.


In [160]:
%%bash
cd ../../Data/Rhoto_IFO0880_3/
blastdbcmd -db Rhoto_IFO0880_3_All_transcripts -entry 900810,900782,900778,866864,950280,900774,900801,518890,950279,946153,979743,863471,900779,950281,946144,861956,518409,946151,879519,900758,946149,900814,900819,946145,889790,979745,900792 \
    > ../Mito/Mito_Transcripts.fa

In [161]:
%%bash
cd ../../Data/Mito
head Mito_Transcripts.fa

>900810 fgenesh1_kg.4_#_70_#_TRINITY_DN3144_c3_g8_i15
GGTTTTGGCAACTACTTAGTGCCAGTTCAAATTGGGGCGCCTGATATGGCCTTCCCACGTCTTAATAACATTAGCTTTTG
GCTACTGCCGCCGTCACTAGTGCTGCTGCTTGCAAGTGCACTAGTCGAACAGGGTGCAGGTACCGGCTGGACTATTTACC
CACCCCTGTCGGGCATTCAAAGCCATTCAGGTGGGTCGGTAGACTTAGCAATTTTTAGCCTGCACCTGTCGGGCATTAGT
AGTATGCTAGGTGCAATGAACTTCATTACTACAATTTTGAACATGCGTGCCCCTGGCATGTCAATGCACAAGCTGCCGCT
TTTTTGCTGGGCAATTTTTATTACAGCAATTTTGCTGCTGCTGTCATTGCCAGTACTAGCGAGGAGCCATTACAATGTTA
CTAACGGACCGTAACTTCAACACGTCCTTTTACGACCCTGCAGGCGGGGGTGACCCAATTTTATTCCAACACCTGTTCTG
GTTTTTTGGTCACCCTGAAGTCTACATTTTAATCATTCCTGCCTTTGGCATTGTGTCGCATGTAGTTGCAACCTTTGCTG
GTAAGCCCGTGTTTGGGTACCTAGGTATGGTCTATGCAATGTTCAGCATTGGCATTTTAGGCTTTATTGTATGTTAGCAC
CACATGTTTGCCGTGGGGCTAGACGTTGACACACGAGCCTACTTCACAGCAGCCACAATGATTATTGCTCGTCCCACAGG


In [172]:
%%bash
cd ../../Data/Mito
blastn -query Mito_Transcripts.fa -subject ../Rhoto_IFO0880_4/Rhoto_IFO0880_4_MitoAssemblyScaffolds.fasta \
    > Mito_Transcripts_to_IFO0880_4.txt

In [188]:
[(x, mapping[x]) for x in targets]
for k, v in Mito_proteinId.items():
    print(k)
    for k2, v2 in sorted(v.items(), key=lambda kv: kv[1], reverse=True):
        print(k2, mapping[k2], v2)

cob
979592 979743 655.721
950128 950279 436.87
900628 900779 159.511
900627 900778 121.25
979594 979745 106.666
cox1
950130 950281 1626.5469999999998
900659 900810 328.382
cox2
950130 950281 1083.48
518258 518409 339.301
cox3
946002 946153 903.938
889639 889790 269.195
900631 900782 170.958
atp9
900663 900814 432.82000000000005
nad1
900641 900792 280.145
950129 950280 58.861
nad3
879368 879519 56.989
863320 863471 52.874
946000 946151 50.515
nad4
900607 900758 279.053
nad4L
945993 946144 72.093
866713 866864 66.279
861805 861956 63.953
518739 518890 55.422
nad5
900668 900819 247.29200000000003
945994 946145 53.959
atp6
900623 900774 51.984
945998 946149 51.634
atp8
900650 900801 110.42099999999999


In [173]:
%%bash
cd ../../Data/Mito
cat Mito_Transcripts_to_IFO0880_4.txt

BLASTN 2.6.0+


Reference: Zheng Zhang, Scott Schwartz, Lukas Wagner, and Webb
Miller (2000), "A greedy algorithm for aligning DNA sequences", J
Comput Biol 2000; 7(1-2):203-14.



Database: User specified sequence set (Input:
/users/joonhoonkim/Data/R_toruloides/Rhoto_IFO0880_4/Rhoto_IFO0880_4
_MitoAssemblyScaffolds.fasta).
           1 sequences; 110,369 total letters



Query= 900810 fgenesh1_kg.4_#_70_#_TRINITY_DN3144_c3_g8_i15

Length=1983
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value

  scaffold_1_mito                                                     2135    0.0  


> scaffold_1_mito
Length=110369

 Score = 2135 bits (1156),  Expect = 0.0
 Identities = 1156/1156 (100%), Gaps = 0/1156 (0%)
 Strand=Plus/Minus

Query  828    TAACAGGTGTCGTACTTGCAAATGCTAGTCTTGACGTTGCAATTCACGATAGTACCGTGT  887
              ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||

In [176]:
%%bash
cd ../../Data/Mito
bp_search2gff --input Mito_Transcripts_to_IFO0880_4.txt --addid --version 3 --type hit

##gff-version 3
scaffold_1_mito	BLASTN	similarity	93917	95072	1156	-	0	ID=900810;Target=Sequence:900810 828 1983
scaffold_1_mito	BLASTN	similarity	101569	101768	200	-	0	ID=900810;Target=Sequence:900810 253 452
scaffold_1_mito	BLASTN	similarity	96728	96859	132	-	0	ID=900810;Target=Sequence:900810 502 633
scaffold_1_mito	BLASTN	similarity	95443	95565	120	-	0	ID=900810;Target=Sequence:900810 710 832
scaffold_1_mito	BLASTN	similarity	93859	94022	119	-	0	ID=900810;Target=Sequence:900810 1820 1983
scaffold_1_mito	BLASTN	similarity	102656	102766	111	-	0	ID=900810;Target=Sequence:900810 142 252
scaffold_1_mito	BLASTN	similarity	102966	103064	96	-	0	ID=900810;Target=Sequence:900810 46 144
scaffold_1_mito	BLASTN	similarity	93975	94080	82	-	0	ID=900810;Target=Sequence:900810 1878 1983
scaffold_1_mito	BLASTN	similarity	96051	96127	77	-	0	ID=900810;Target=Sequence:900810 633 709
scaffold_1_mito	BLASTN	similarity	103714	103758	45	-	0	ID=900810;Target=Sequence:900810 1 45
scaffold_1_mito	BLASTN	simil

In [177]:
%%bash
cd ../../Data/Mito
bp_search2gff --input Mito_Transcripts_to_IFO0880_4.txt --addid --version 3 --type hit --match

##gff-version 3
scaffold_1_mito	BLASTN	similarity	93917	95072	1156	-	0	Parent=900810;Target=Sequence:900810 828 1983
scaffold_1_mito	BLASTN	similarity	101569	101768	200	-	0	Parent=900810;Target=Sequence:900810 253 452
scaffold_1_mito	BLASTN	similarity	96728	96859	132	-	0	Parent=900810;Target=Sequence:900810 502 633
scaffold_1_mito	BLASTN	similarity	95443	95565	120	-	0	Parent=900810;Target=Sequence:900810 710 832
scaffold_1_mito	BLASTN	similarity	93859	94022	119	-	0	Parent=900810;Target=Sequence:900810 1820 1983
scaffold_1_mito	BLASTN	similarity	102656	102766	111	-	0	Parent=900810;Target=Sequence:900810 142 252
scaffold_1_mito	BLASTN	similarity	102966	103064	96	-	0	Parent=900810;Target=Sequence:900810 46 144
scaffold_1_mito	BLASTN	similarity	93975	94080	82	-	0	Parent=900810;Target=Sequence:900810 1878 1983
scaffold_1_mito	BLASTN	similarity	96051	96127	77	-	0	Parent=900810;Target=Sequence:900810 633 709
scaffold_1_mito	BLASTN	similarity	103714	103758	45	-	0	Parent=900810;Target=Sequence:

In [178]:
%%bash
cd ../../Data/Mito
bp_search2gff --input Mito_Transcripts_to_IFO0880_4.txt --addid --version 3 --type hit --match \
    > Mito_Transcripts_to_IFO0880_4.gff3

In [191]:
sorted(Mito_proteinId.items())

[('atp6', {'900623': 51.984, '945998': 51.634}),
 ('atp8', {'900650': 110.42099999999999}),
 ('atp9', {'900663': 432.82000000000005}),
 ('cob',
  {'979592': 655.721,
   '979594': 106.666,
   '950128': 436.87,
   '900628': 159.511,
   '900627': 121.25}),
 ('cox1', {'900659': 328.382, '950130': 1626.5469999999998}),
 ('cox2', {'950130': 1083.48, '518258': 339.301}),
 ('cox3', {'946002': 903.938, '889639': 269.195, '900631': 170.958}),
 ('nad1', {'900641': 280.145, '950129': 58.861}),
 ('nad3', {'863320': 52.874, '879368': 56.989, '946000': 50.515}),
 ('nad4', {'900607': 279.053}),
 ('nad4L',
  {'518739': 55.422, '945993': 72.093, '861805': 63.953, '866713': 66.279}),
 ('nad5', {'900668': 247.29200000000003, '945994': 53.959})]

atp6 RTO3_900623 Q0085 ATPeF0A, MTATP6, ATP6; F-type H+-transporting ATPase subunit a  
atp8 RTO3_879280 Q0080 ATPase, F0 complex, subunit 8, mitochondrial, fungal (not final gene model)  
atp9 RTO3_900663 Q0130 ATPeF0C, ATP5G, ATP9; F-type H+-transporting ATPase subunit c  

COB (Q0105, CYTB) RTO3_979594 # initially had it as 946001, but 979594 matches splicing better  

COX1, COX2 (I, II) RTO3_946004  
COX3 (III) RTO3_946002  

RTO3_900641 ND1; NADH-ubiquinone oxidoreductase chain 1  
RTO3_874609 ND3; NADH-ubiquinone oxidoreductase chain 3  
RTO3_945995 ND4; NADH-ubiquinone oxidoreductase chain 4  
RTO3_879512 NADH dehydrogenase subunit 4L and related proteins  
RTO3_879341 NADH dehydrogenase subunit 4L and related proteins  
RTO3_945994 ND5; NADH-ubiquinone oxidoreductase chain 5  
RTO3_946005 ND5; NADH-ubiquinone oxidoreductase chain 5  
RTO3_900622 ND6; NADH-ubiquinone oxidoreductase chain 6  
RTO3_900624 nuoN; NADH-quinone oxidoreductase subunit N